relayer-cli/src/commands/query/channels.rs

```diff
@@ -1,60 +1,184 @@
+use std::fmt::{Debug, Error, Formatter};
 use std::sync::Arc;
 
 use abscissa_core::{Options, Runnable};
+use serde::Serialize;
 use tokio::runtime::Runtime as TokioRuntime;
 
+use ibc::ics04_channel::channel::State;
 use ibc::ics24_host::identifier::{ChainId, PortChannelId};
 use ibc_proto::ibc::core::channel::v1::QueryChannelsRequest;
 use ibc_relayer::chain::{Chain, CosmosSdkChain};
 
+use crate::commands::query::channel_ends::ChannelEnds;
 use crate::conclude::Output;
 use crate::prelude::*;
 
 #[derive(Clone, Command, Debug, Options)]
 pub struct QueryChannelsCmd {
     #[options(free, required, help = "identifier of the chain to query")]
     chain_id: ChainId,
+
+    #[options(help = "identifier of the channel's destination chain", short = "d")]
+    destination_chain: Option<ChainId>,
+
+    #[options(
+        help = "enable verbose output, displaying all client and connection ids",
+        short = "v"
+    )]
+    verbose: bool,
+}
+
+fn run_query_channels(
+    cmd: &QueryChannelsCmd,
+) -> Result<QueryChannelsOutput, Box<dyn std::error::Error>> {
+    debug!("Options: {:?}", cmd);
+
+    let mut output = vec![];
+    let mut output_verbose = vec![];
+
+    let config = app_config();
+    let chain_id = cmd.chain_id.clone();
+
+    let chain_config = config
+        .find_chain(&chain_id)
+        .ok_or_else(|| format!("chain '{}' not found in configuration file", chain_id))?;
+
+    let rt = Arc::new(TokioRuntime::new()?);
+    let chain = CosmosSdkChain::bootstrap(chain_config.clone(), rt.clone())?;
+    let chain_height = chain.query_latest_height()?;
+
+    let req = QueryChannelsRequest {
+        pagination: ibc_proto::cosmos::base::query::pagination::all(),
+    };
+    let identified_channels = chain.query_channels(req)?;
+
+    for identified_channel in identified_channels {
```

Comment:
I think the body of this for loop should be split into two more functions: one when `verbose` is true and one when `verbose` is false.

Reply:
https://github.com/informalsystems/ibc-rs/pull/1243/commits/4e084a86c8a1ace8ac702ab6e094dd66feeb8bfb